apps/builder/app/builder/builder.tsx

```diff
@@ -307,55 +309,91 @@ export const Builder = ({
     project,
   });
 
+  /**
+   * Prevents Lexical text editor from stealing focus during rendering.
+   * Sets the inert attribute on the canvas body element and disables the text editor.
+   *
+   * This must be done synchronously to avoid the following issue:
+   *
+   * 1. Text editor is in edit state.
+   * 2. User focuses on the builder (e.g., clicks any input).
+   * 3. The text editor blur event triggers, causing a rerender on data change (data saved in onBlur).
+   * 4. Text editor rerenders, stealing focus from the builder.
+   * 5. Inert attribute is set asynchronously, but focus is already lost.
+   *
+   * Synchronous focusing and setInert prevent the text editor from focusing on render.
+   * This cannot be handled inside the canvas because the text editor toolbar is in the builder and focus events in the canvas should be ignored.
+   */
+  const handleFocus = useCallback((event: React.FocusEvent) => {
+    // Ignore toolbar focus events. See the onFocus handler in text-toolbar.tsx
+    if (false === event.defaultPrevented) {
+      canvasApi.setInert();
+      $textEditingInstanceSelector.set(undefined);
+    }
+  }, []);
+
+  /**
+   * Prevent radix to steal focus during editing in the settings panel.
```

Comment:
will need a more detailed comment to 100% understand how to reproduce the situation, for the future self

Reply:
hover on setInert